internal/security/acctz/acctz.go

```diff
@@ -0,0 +1,917 @@
+			Origin: "native",
+		},
+		Replace: []*gnmi.Update{
+			{
+				Path: &gnmi.Path{
+					Elem: []*gnmi.PathElem{
+						{Name: "system"},
+						{Name: "aaa"},
+						{Name: "authorization"},
+						{Name: "role", Key: map[string]string{"rolename": failRoleName}},
+					},
+				},
+				Val: &gnmi.TypedValue{
+					Value: &gnmi.TypedValue_JsonIetfVal{
+						JsonIetfVal: failRoleData,
+					},
+				},
+			},
+		},
+	}
+}
+
+// SetupGrpcUsers Setup users for grpc-based acctz tests.
+func SetupGrpcUsers(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	successUser := auth.GetOrCreateUser(successUsername)
+	successUser.SetRole(oc.AaaTypes_SYSTEM_DEFINED_ROLES_SYSTEM_ROLE_ADMIN)
+	auth.GetOrCreateUser(failUsername)
+	ondatragnmi.Update(t, dut, ondatragnmi.OC().System().Aaa().Authentication().Config(), auth)
+	setupUserPassword(t, dut, successUsername, successPassword)
+}
+
+// SetupSSHUsers Setup users for ssh-based acctz tests.
+func SetupSSHUsers(t *testing.T, dut *ondatra.DUTDevice) {
+	var SetRequest *gnmi.SetRequest
+
+	// Create failure cli role in native.
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
+		SetRequest = nokiaFailCliRole(t)
+	}
+
+	gnmiClient := dut.RawAPIs().GNMI(t)
+	if _, err := gnmiClient.Set(context.Background(), SetRequest); err != nil {
+		t.Fatalf("Unexpected error configuring role: %v", err)
+	}
+
+	// Configure users.
+	auth := &oc.System_Aaa_Authentication{}
+	successUser := auth.GetOrCreateUser(successUsername)
+	successUser.SetRole(oc.AaaTypes_SYSTEM_DEFINED_ROLES_SYSTEM_ROLE_ADMIN)
+	failUser := auth.GetOrCreateUser(failUsername)
+	failUser.SetRole(oc.UnionString(failRoleName))
+	ondatragnmi.Update(t, dut, ondatragnmi.OC().System().Aaa().Authentication().Config(), auth)
+	setupUserPassword(t, dut, successUsername, successPassword)
+	setupUserPassword(t, dut, failUsername, failPassword)
+}
+
+func getGrpcTarget(t *testing.T, dut *ondatra.DUTDevice, service introspect.Service) string {
+	dialTarget := introspect.DUTDialer(t, dut, service).DialTarget
+	resolvedTarget, err := net.ResolveTCPAddr("tcp", dialTarget)
+	if err != nil {
+		t.Fatalf("Failed resolving %s target %s", service, dialTarget)
+	}
+	t.Logf("Target for %s service: %s", service, resolvedTarget)
+	return resolvedTarget.String()
+}
+
+func getSSHTarget(t *testing.T, dut *ondatra.DUTDevice) string {
+	var serviceDUT interface {
+		Service(string) (*tpb.Service, error)
+	}
+
+	var target string
+	err := binding.DUTAs(dut.RawAPIs().BindingDUT(), &serviceDUT)
+	if err != nil {
+		t.Log("DUT does not support `Service` function, will attempt to resolve dut name field.")
+
+		// Suppose ssh could be not 22 in some cases but don't think this is exposed by introspect.
+		dialTarget := fmt.Sprintf("%s:%d", dut.Name(), defaultSSHPort)
+		resolvedTarget, err := net.ResolveTCPAddr("tcp", dialTarget)
+		if err != nil {
+			t.Fatalf("Failed resolving ssh target %s", dialTarget)
+		}
+		target = resolvedTarget.String()
+	} else {
```

Comment:
chances are this else isn't necessary? (or could be made to be not necessary)

Reply:
Yes, this can be made to be not necessary by returning 2 times from the func. I was looking to log the target string once before returning hence the else.